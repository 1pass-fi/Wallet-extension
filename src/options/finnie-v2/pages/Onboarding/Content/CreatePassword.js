import React, { useState, useContext } from 'react'

import WelcomeBackgroundTop from 'img/v2/onboarding/welcome-background-top.svg'
import WelcomeBackgroundBottom from 'img/v2/onboarding/welcome-background-bottom.svg'

import Button from 'finnie-v2/components/Button'
import InputField from 'finnie-v2/components/InputField'
import CheckBox from 'finnie-v2/components/CheckBox'

import { URL } from 'constants/koiConstants'

import { OnboardingContext } from '../onboardingContext'

const CreatePassword = ({ step, setStep }) => {
  const {
    password,
    setPassword,
    confirmPassword,
    setConfirmPassword,
    isValidPassword,
    passwordErrorMessage
  } = useContext(OnboardingContext)

  const [isAcceptTermService, setIsAcceptTermService] = useState(false)

  return (
    <div className="w-2/3 flex flex-col text-white">
      <WelcomeBackgroundTop className="absolute top-0 right-0" />
      <WelcomeBackgroundBottom className="absolute bottom-0 left-0" />
      <div className="font-normal text-base leading-6 text-left" style={{ width: '265px' }}>
        Create a password to secure Finnie.
      </div>
      <InputField
        className="my-5 h-5.25"
        label="New Password"
        value={password}
        setValue={(e) => setPassword(e.target.value)}
        required={true}
        name="password"
        description="Secure passwords have at least 8 characters and include uppercase & lowercase letters, numbers, and special characters (e.g. !@#$%)."
        placeholder=""
        uppercase={false}
        password={true}
      />
      <InputField
        className="my-12 h-5.25"
        label="Confirm Password"
        value={confirmPassword}
        setValue={(e) => setConfirmPassword(e.target.value)}
        required={true}
        name="password"
        placeholder=""
        // description="Passwords do not match"
        uppercase={false}
        password={true}
      />
      <div className="flex items-center justify-center">
        <CheckBox
          checked={isAcceptTermService}
          onClick={() => setIsAcceptTermService((prev) => !prev)}
          theme="dark"
          className="w-5 h-5"
        />
        <div className="w-full flex ml-2.25">
          <div onClick={() => setIsAcceptTermService((prev) => !prev)}>I agree with the&nbsp;</div>
          <a target="_blank" href={URL.TERM_OF_SERVICE} className="text-success underline">
            Terms of Service
          </a>
        </div>
        <div>{passwordErrorMessage}</div>
      </div>
      <Button
        style={{ width: '240px', height: '42px' }}
        className="mt-6 text-base rounded mx-auto z-10"
        variant="white"
        text="Log In"
        onClick={() => setStep(step + 1)}
      />
    </div>
  )
}

export default CreatePassword
