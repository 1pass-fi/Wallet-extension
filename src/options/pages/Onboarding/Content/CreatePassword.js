import React, { useContext, useMemo, useState } from 'react'
import { useSelector } from 'react-redux'
import { useLocation } from 'react-router-dom'
import clsx from 'clsx'
import { URL } from 'constants/koiConstants'
import WelcomeBackgroundBottom from 'img/v2/onboarding/welcome-background-bottom.svg'
import WelcomeBackgroundTop from 'img/v2/onboarding/welcome-background-top.svg'
import isEmpty from 'lodash/isEmpty'
import Button from 'options/components/Button'
import CheckBox from 'options/components/CheckBox'
import InputField from 'options/components/InputField'

import { VALIDATE_ERROR_MESSAGE } from '../../../../constants/koiConstants'
import { OnboardingContext } from '../onboardingContext'

const CreatePassword = ({ step, setStep }) => {
  const location = useLocation()
  const {
    password,
    setPassword,
    confirmPassword,
    setConfirmPassword,
    isValidPassword,
    passwordErrorMessage,
    verifyPassword
  } = useContext(OnboardingContext)

  const [isAcceptTermService, setIsAcceptTermService] = useState(false)
  const [termServiceMessage, setTermServiceMessage] = useState('')

  const [wrongPasswordMessage, setWrongPasswordMessage] = useState('')
  const [isClickContinue, setIsClickContinue] = useState(false)

  const onboardingPath = useSelector((state) => state.onboarding.path)
  const accounts = useSelector((state) => state.accounts)

  const onClickContinue = async () => {
    if (!isEmpty(accounts)) {
      const isCorrectPassword = await verifyPassword()
      if (isCorrectPassword) {
        if (onboardingPath === '/create-wallet') {
          setStep(2)
        } else if (onboardingPath === '/import-wallet') {
          setStep(10)
        } else if (onboardingPath === '/upload-json') {
          setStep(13)
        } else {
          setStep(1)
        }
      } else {
        setWrongPasswordMessage(chrome.i18n.getMessage('incorrectPassword'))
      }
      return
    }

    if (!isAcceptTermService) {
      setTermServiceMessage(chrome.i18n.getMessage('pleaseAcceptTheTof'))
    } else {
      setTermServiceMessage('')
      if (isValidPassword) {
        setStep(step + 1)
      } else {
        setIsClickContinue(true)
      }
    }
  }

  const handleKeyDown = async (e) => {
    setIsClickContinue(false)
    if (e.keyCode === 13) {
      onClickContinue()
    }
  }

  console.log('location.pathname--------', location.pathname)

  return (
    <div
      data-testid="CreatePassword"
      className={clsx(
        'w-2/5 flex flex-col text-white self-start ml-23',
        location.pathname === '/upload-json' && 'ml-60'
      )}
      style={{ minWidth: '240px' }}
    >
      <WelcomeBackgroundTop className={clsx('welcome-bg-top')} />
      <WelcomeBackgroundBottom className={clsx('welcome-bg-bottom')} />
      <div className="font-normal text-base leading-6 text-left z-10">
        {isEmpty(accounts)
          ? chrome.i18n.getMessage('createAPasswordToSecureFinnie')
          : chrome.i18n.getMessage('reEnterYourFinniePassword')}
      </div>
      <InputField
        className="mt-5 z-10"
        label={
          isEmpty(accounts)
            ? chrome.i18n.getMessage('newPassword')
            : chrome.i18n.getMessage('password')
        }
        value={password}
        setValue={(e) => {
          setWrongPasswordMessage('')
          setPassword(e.target.value)
        }}
        required={true}
        name="password"
        errorFinnie={
          isEmpty(accounts) &&
          !isEmpty(password) &&
          isClickContinue &&
          (passwordErrorMessage === chrome.i18n.getMessage('passwordNotMeetRequirements') ||
            passwordErrorMessage === chrome.i18n.getMessage('notEnoughCharacters'))
            ? chrome.i18n.getMessage('securePasswordsHaveAtLeast8Characters')
            : !isEmpty(wrongPasswordMessage)
              ? wrongPasswordMessage
              : ''
        }
        placeholder=""
        uppercase={false}
        passwordFinnie={true}
        autoFocus={true}
        onKeyDown={(e) => handleKeyDown(e)}
        id={'new-password'}
        errorId={'error-new-password'}
      />
      {isEmpty(accounts) && (
        <InputField
          className="mt-7 z-10"
          label={chrome.i18n.getMessage('confirmPassword')}
          value={confirmPassword}
          setValue={(e) => setConfirmPassword(e.target.value)}
          required={true}
          name="password"
          placeholder=""
          errorFinnie={
            isClickContinue &&
            passwordErrorMessage === chrome.i18n.getMessage('passwordDoesNotMatch') &&
            passwordErrorMessage
          }
          uppercase={false}
          passwordFinnie={true}
          onKeyDown={(e) => handleKeyDown(e)}
          id={'confirm-password'}
          errorId={'error-confirm-password'}
        />
      )}
      {isEmpty(accounts) && (
        <div className="mt-5 flex items-center justify-center z-10">
          <CheckBox
            checked={isAcceptTermService}
            onClick={() => setIsAcceptTermService((prev) => !prev)}
            theme="dark"
            className="w-5 h-5"
            data-testid="new-password-tos"
            role="checkbox"
          />
          <div className="flex ml-2.25 font-normal text-sm leading-6 z-10">
            <div onClick={() => setIsAcceptTermService((prev) => !prev)}>
              {chrome.i18n.getMessage('iAgreeWithThe')}&nbsp;
            </div>
            <a
              target="_blank"
              href={URL.TERM_OF_SERVICE}
              className="text-turquoiseBlue underline"
              data-testid="tos-link"
            >
              {chrome.i18n.getMessage('termsOfService')}
            </a>
          </div>
        </div>
      )}
      <div
        data-testid="tos-error-message"
        className="text-red-finnie text-xs font-normal mt-2 text-center"
      >
        {termServiceMessage}
      </div>
      <Button
        style={{ width: '240px', height: '42px' }}
        className="mt-3.5 rounded mx-auto z-10"
        variant="white"
        text={chrome.i18n.getMessage('logIn')}
        onClick={() => onClickContinue()}
        id="log-in-button"
      />
    </div>
  )
}

export default CreatePassword
