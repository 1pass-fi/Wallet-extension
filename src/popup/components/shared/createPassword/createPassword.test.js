import '@babel/polyfill'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import React from 'react'

import CreatePassword from '.'

describe('Test for CreatePassword component', () => {
  describe('Render without crashing', () => {
    it('renders correctly', () => {
      const { container } = render(<CreatePassword />)
      expect(container).toMatchSnapshot()
    })
  })
})
