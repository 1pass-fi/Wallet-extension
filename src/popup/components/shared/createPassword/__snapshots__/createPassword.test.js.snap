// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for CreatePassword component Render without crashing renders correctly 1`] = `
<div>
  <div
    class="create-password"
  >
    <div
      class="fields"
    >
      <div
        class="input-field "
      >
        <div
          class="label"
        >
          New password
        </div>
        <div
          class="input-wrapper"
        >
          <input
            class="input"
            name="pwd"
            placeholder="Make it unique (min. 8 characters)"
            type="password"
            value=""
          />
          <svg
            class="toggle-display"
          />
        </div>
      </div>
      <div
        class="input-field "
      >
        <div
          class="label"
        >
          Confirm password
        </div>
        <div
          class="input-wrapper"
        >
          <input
            class="input"
            name="pwdConfirm"
            placeholder=""
            type="password"
            value=""
          />
          <svg
            class="toggle-display"
          />
        </div>
      </div>
    </div>
    <div
      class="term-service"
    >
      <div
        class="checkbox"
      >
        <input
          name="checkbox"
          type="checkbox"
          value="false"
        />
      </div>
      <label>
        I agree with the 
        <a
          href="#"
        >
          Terms of Service
        </a>
      </label>
    </div>
    <div
      class="button"
    >
      <button
        class="button-shared button-filled "
      >
        Import Wallet
      </button>
    </div>
  </div>
</div>
`;
